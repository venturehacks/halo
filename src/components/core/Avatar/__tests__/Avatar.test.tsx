import { render } from 'enzyme';
import React from 'react';

import { Avatar } from '../index';

const sharedProps = {
  imageUrl: 'https://avatars1.githubusercontent.com/u/194885?s=120&v=6',
  name: 'Drew',
};

describe('Avatar', () => {
  test('smoke', () => {
    const component = render(<Avatar {...sharedProps} size="large" />);

    expect(component.html()).toContain('<img');
  });

  describe('snapshots', () => {
    describe('size', () => {
      test(`xxxsmall`, () => {
        const component = render(<Avatar {...sharedProps} size="xxxsmall" />);

        expect(component).toMatchSnapshot();
      });

      test(`medium [default]`, () => {
        const component = render(<Avatar {...sharedProps} size="medium" />);

        expect(component).toMatchSnapshot();
      });

      test(`large`, () => {
        const component = render(<Avatar {...sharedProps} size="large" />);

        expect(component).toMatchSnapshot();
      });
    });

    describe('shape', () => {
      test(`circle [default]`, () => {
        const component = render(<Avatar {...sharedProps} size="medium" />);

        expect(component).toMatchSnapshot();
      });
      test(`square`, () => {
        const component = render(
          <Avatar {...sharedProps} shape="square" size="medium" />,
        );

        expect(component).toMatchSnapshot();
      });
    });

    test(`badge label`, () => {
      const component = render(
        <Avatar {...sharedProps} badge="Test Badge" size="large" />,
      );

      expect(component.text()).toContain('Test Badge');
      expect(component).toMatchSnapshot();
    });

    test(`badgeColor`, () => {
      const component = render(
        <Avatar
          {...sharedProps}
          badge="Purple"
          badgeColor="purple"
          size="large"
        />,
      );

      expect(component.text()).toContain('Purple');
      expect(component).toMatchSnapshot();
    });

    describe(`badgeShape`, () => {
      test(`circle [default]`, () => {
        const component = render(
          <Avatar {...sharedProps} badge="Test Badge" size="large" />,
        );

        expect(component).toMatchSnapshot();
      });

      test(`square`, () => {
        const component = render(
          <Avatar
            {...sharedProps}
            badge="Test Badge"
            badgeShape="square"
            size="large"
          />,
        );

        expect(component).toMatchSnapshot();
      });
    });
  });
});
