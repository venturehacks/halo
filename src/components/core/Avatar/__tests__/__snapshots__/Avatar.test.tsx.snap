// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar snapshots badge label 1`] = `
<div
  class="component large circle"
>
  <img
    alt="Avatar for Drew"
    class="avatar circle"
    src="https://avatars1.githubusercontent.com/u/194885?s=120&v=6"
  />
  <div
    class="component gray circle"
  >
    Test Badge
  </div>
</div>
`;

exports[`Avatar snapshots badgeColor 1`] = `
<div
  class="component large circle"
>
  <img
    alt="Avatar for Drew"
    class="avatar circle"
    src="https://avatars1.githubusercontent.com/u/194885?s=120&v=6"
  />
  <div
    class="component purple circle"
  >
    Purple
  </div>
</div>
`;

exports[`Avatar snapshots badgeShape circle [default] 1`] = `
<div
  class="component large circle"
>
  <img
    alt="Avatar for Drew"
    class="avatar circle"
    src="https://avatars1.githubusercontent.com/u/194885?s=120&v=6"
  />
  <div
    class="component gray circle"
  >
    Test Badge
  </div>
</div>
`;

exports[`Avatar snapshots badgeShape square 1`] = `
<div
  class="component large circle"
>
  <img
    alt="Avatar for Drew"
    class="avatar circle"
    src="https://avatars1.githubusercontent.com/u/194885?s=120&v=6"
  />
  <div
    class="component gray square"
  >
    Test Badge
  </div>
</div>
`;

exports[`Avatar snapshots shape circle [default] 1`] = `
<div
  class="component medium circle"
>
  <img
    alt="Avatar for Drew"
    class="avatar circle"
    src="https://avatars1.githubusercontent.com/u/194885?s=120&v=6"
  />
</div>
`;

exports[`Avatar snapshots shape square 1`] = `
<div
  class="component medium square"
>
  <img
    alt="Avatar for Drew"
    class="avatar square"
    src="https://avatars1.githubusercontent.com/u/194885?s=120&v=6"
  />
</div>
`;

exports[`Avatar snapshots size large 1`] = `
<div
  class="component large circle"
>
  <img
    alt="Avatar for Drew"
    class="avatar circle"
    src="https://avatars1.githubusercontent.com/u/194885?s=120&v=6"
  />
</div>
`;

exports[`Avatar snapshots size medium [default] 1`] = `
<div
  class="component medium circle"
>
  <img
    alt="Avatar for Drew"
    class="avatar circle"
    src="https://avatars1.githubusercontent.com/u/194885?s=120&v=6"
  />
</div>
`;

exports[`Avatar snapshots size xxxsmall 1`] = `
<div
  class="component xxxsmall circle"
>
  <img
    alt="Avatar for Drew"
    class="avatar circle"
    src="https://avatars1.githubusercontent.com/u/194885?s=120&v=6"
  />
</div>
`;
